pystac/extensions/ext.py

```diff
@@ -406,7 +406,7 @@ def file(self) -> FileExtension[Asset]:
 
     @property
     def mlm(self) -> AssetGeneralMLMExtension[Asset] | AssetDetailedMLMExtension:
-        if "mlm:name" in self.stac_object.extra_fields:
+        if "mlm:architecture" in self.stac_object.extra_fields:
```

Comment:
What's the reason for this change? It's not clear to me from reading the extension description why the presence of this field should trigger different behavior.